crates/script/src/simulate.rs

```diff
@@ -112,10 +112,14 @@ impl PreSimulationState {
         // Executes all transactions from the different forks concurrently.
         let futs = transactions
             .into_iter()
-            .map(|transaction| async {
+            .map(|mut transaction| async {
                 let mut runner = runners.get(&transaction.rpc).expect("invalid rpc url").write();
+                let tx = transaction.tx_mut();
+
+                if let Some(authorization_list) = tx.authorization_list() {
+                    runner.executor.set_delegation(&authorization_list)?;
```

Comment:
looks like this `set_delegation` call would set the authorization list not just for this transaction but for all transactions after it as well?

Reply:
Based on my tests I think the current logic will use the delegation when the EOA is called, but not on calls to any other accounts
```
vm.attachDelegation(impl, alice, v, r, s);
SimpleDelegateContract(alice).execute(calls1);
// works the same as call1 because we have not changed/removed the delegation
SimpleDelegateContract(alice).execute(calls2); 

token.mint(100, bob); // does not use the delegation
```

Is that how you envisage the logic to work, or would you prefer to explicitly revoke the delegation after the first transaction? Let me know if I'm missing something as well.

Reply:
if an authorization is applied to transaction, then it is persisted in database and there's no need to keep this delegation in `TxEnv` or to attach to any other transactions

storing it in TxEnv for all other transactions would result in gas cost for all subsequent calls including the cost of authorization which is not something we want

Reply:
Got it, thanks. I've added logic to set `authorization_list` to `None` if `active_delegation` is `None`, and added a test to assert that attaching another delegation to a tx in the same bundle still switches out the bytecode on the EOA.

Reply:
I misunderstood - we need to clear `TxEnv.authorization_list` between transactions. would adding `unset_delegation()` to clear it when `tx.authorization_list()` is `None` work? Or is there a better approach?

Reply:
imo we could just handle this differently, similar to how to/data/value are set, without touching the global executor env.

`ScriptRunner::simulate` could take authorization list as an additional argument, and then we can either
1. Manually call `Executor::build_test_env`, set authorization list to the created temporary TxEnv, and pass it to `call_with_env` just as `Executor::call_raw` does
2. Or add a helper method `Executor::call_raw_with_authorization` which would be same as `call_raw` but also accept authorization list

Reply:
Got it, thanks for the clarification. I've pushed [a commit](https://github.com/foundry-rs/foundry/pull/9236/commits/2a30b429a2fa8f2a9e32277374f2bcb4502a2c09) which uses your option 2.